test/runner.js

```diff
@@ -10,87 +9,133 @@ require( 'babel-register' );
 const debug = require( 'debug' )( 'test-runner' ),
 	glob = require( 'glob' ),
 	Mocha = require( 'mocha' ),
-	path = require( 'path' ),
 	program = require( 'commander' ),
+	watchFile = require( 'fs' ).watchFile,
+	fs = require( 'fs' ),
+	isEmpty = require( 'lodash/isEmpty' ),
 	chalk = require( 'chalk' );
 
 /**
  * Internal dependencies
  */
-const boot = require( './boot-test' ),
-	setup = require( './setup' );
+const boot = require( './boot-test' );
 
 program
 	.usage( '[options] [files]' )
 	.option( '-R, --reporter <name>', 'specify the reporter to use', 'spec' )
 	.option( '-t, --node-total <n>', 'specify the node total to use', parseInt )
 	.option( '-i, --node-index <n>', 'specify the node index to use', parseInt )
-	.option( '-g, --grep <pattern>', 'only run tests matching <pattern>' );
+	.option( '-g, --grep <pattern>', 'only run tests matching <pattern>' )
+	.option( '-w, --watch', 'watch files for changes' );
 
 program.name = 'runner';
 
 program.parse( process.argv );
 
-const mocha = new Mocha( {
-	ui: 'bdd',
-	reporter: program.reporter
-} );
+let getTestFiles = () => {
+	let testFiles = program.args;
+	if ( isEmpty( program.args ) ) {
```

Comment:
More food for thought since it doesn't bother me much, but seems `! program.args.length` would work just as well here. `isEmpty` is nice in that it's human-readable, but knowing that it supports both objects and arrays makes me a bit more uncertain what it is you're testing.